### ISSUE TITLE
[MNG-7362] DefaultArtifactResolver has spurious "Failure detected" INFO log

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8066]
**[Frederic Lionello](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER281732)** opened **[MNG-7362](https://issues.apache.org/jira/browse/MNG-7362?redirect=false)** and commented

The DefaultArtifactResolver may issue an INFO log "Failure detected.", without any additional context.

This log was introduced in the changes of for MNG-6057 (see below for the file change and context) and, being an INFO message with "Failure" text, does seem like a debug leftover. On occasions, a maven command will display this log message while still being successful, leaving users wondering what has really happened.

```java
$ git show 51cc76c32625be2f807dcf2ffbeb085984729b57  maven-compat/src/main/java/org/apache/maven/artifact/resolver/DefaultArtifactResolver.java
commit 51cc76c32625be2f807dcf2ffbeb085984729b57
Author: Karl Heinz Marbaise <khmarbaise@apache.org>
Date:   Tue Sep 29 11:46:48 2015 +0200    [MNG-6090] CI friendly properties break submodule builds
    [MNG-6057] Problem with CI friendly usage of ${..} reactor order is changed
     o Based on the missing replacement of the versions ${revision}
       ${changelist} or ${sha1} within the parent element the order
       of the reactor changes.
    [MNG-5895] Problem with CI friendly usage of ${..} which is already
    defined via property in pom file.diff --git a/maven-compat/src/main/java/org/apache/maven/artifact/resolver/DefaultArtifactResolver.java b/maven-compat/src/main/java/org/apache/maven/artifact/resolver/DefaultArtifactResolver.java
index fc154cb8a..915ee725f 100644
--- a/maven-compat/src/main/java/org/apache/maven/artifact/resolver/DefaultArtifactResolver.java
+++ b/maven-compat/src/main/java/org/apache/maven/artifact/resolver/DefaultArtifactResolver.java
@@ -376,7 +376,7 @@ public ArtifactResolutionResult resolve( ArtifactResolutionRequest request )
         ArtifactFilter resolutionFilter = request.getResolutionFilter();
         RepositorySystemSession session = getSession( request.getLocalRepository() );-        // TODO hack because metadata isn't generated in m2e correctly and i want to run the maven i have in the
+        // TODO: hack because metadata isn't generated in m2e correctly and i want to run the maven i have in the
         // workspace
         if ( source == null )
         {
@@ -506,6 +506,7 @@ public ArtifactResolutionResult resolve( ArtifactResolutionRequest request )
         if ( result.hasMetadataResolutionExceptions() || result.hasVersionRangeViolations()
             || result.hasCircularDependencyExceptions() )
         {
+            logger.info( "Failure detected." );
             return result;
         } 
```

---

**Affects:** 3.5.0-beta-1, 3.6.3

**Issue Links:**
- [MNG-6057](https://issues.apache.org/jira/browse/MNG-6057) Problem with CI friendly usage of ${..} reactor order is changed
 (_**"is caused by"**_)

**Backported to:** [4.0.0-alpha-2](https://github.com/apache/maven/milestone/113?closed=1), [3.8.5](https://github.com/apache/maven/milestone/106?closed=1)

0 votes, 6 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

That is weird since no context information is given.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I think this won't be fixed because `maven-compat` will be removed in the future. [Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), [Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas), any reason to take here a look?

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

So here's the weird thing: even though it is in the (maven2) compat jar, it is frankly still being used. But yes, I expect this jar to be removed and that is has been replaced with something else (that might have the same issue?)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise) it seems it's a personal debug message that you inadvertently committed, isn't it?
https://github.com/apache/maven/commit/51cc76c32625be2f807dcf2ffbeb085984729b57#diff-ad93deb8d1063af1c57044fb0d8387e6182327ee1ac73fc8b83c830b9045114cR509

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

The log info seemed something I simply overlooked.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

thank you, then it's "just delete"

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [7a772fecc02082834ddbe5423376b88d09e38814](https://gitbox.apache.org/repos/asf?p=maven.git&a=commit&h=7a772fecc02082834ddbe5423376b88d09e38814) in master and with [84cbe1ab2e22a12c2c108d1a0e5233f875475fff](https://gitbox.apache.org/repos/asf?p=maven.git&a=commit&h=84cbe1ab2e22a12c2c108d1a0e5233f875475fff) in 3.8.x.
